src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,160 @@
+
+  render() {
+    const { page: pageFromProps, pageSize: pageSizeFormProps, total, pageSizeOptions = [], showSizeChanger } = this.props;
+    const { page: pageFromState, pageSize: pageSizeFromState } = this.state;
+
+    const page = pageFromProps || pageFromState;
+    const pageSize = pageSizeFormProps || pageSizeFromState;
+
+    const numberOfPages = Math.ceil(total / pageSize);
+
+    const leftSide = Math.min(page - VISIBLE_RANGE, numberOfPages - VISIBLE_RANGE * 2);
+    const rightSide = page + VISIBLE_RANGE;
+
+    const showLeftMore = leftSide > 2;
+    const showRightMore = rightSide < numberOfPages - 1;
+
+    const start = leftSide > 1 ? leftSide : 1;
+
+    const pages = [...Array(VISIBLE_RANGE * 2 + 1).keys()].slice(0, numberOfPages);
+
+    return (
+      <PaginationTag tagName="div">
+        <Button size="sm" className={ css`flex-shrink: 0;` } onClick={ this.onChange(page - 1) } color="neutral" variant="outlined" squared disabled={ page <= 1 }>
+          <Icon name="ChevronLeft" size="xs" />
+        </Button>
+
+        <PaginationItemsTag>
+          <If condition={ leftSide > 1 }>
+            <PaginationItemTag active={ page === 1 } onClick={ this.onChange(1) }>1</PaginationItemTag>
+          </If>
+
+          <If condition={ showLeftMore }>
+            <PaginationItemTag disabled>
+              <Icon name="More" />
+            </PaginationItemTag>
+          </If>
+
+          {
+            pages.map((index) => (
+              <PaginationItemTag key={ index + start } active={ index + start === page } onClick={ this.onChange(index + start) }>
+                { index + start }
+              </PaginationItemTag>
+            ))
+          }
+
+          <If condition={ showRightMore }>
+            <PaginationItemTag disabled>
+              <Icon name="More" />
+            </PaginationItemTag>
+          </If>
+
+          <If condition={ rightSide < numberOfPages }>
+            <PaginationItemTag active={ page === numberOfPages } onClick={ this.onChange(numberOfPages) }>{ numberOfPages }</PaginationItemTag>
+          </If>
+        </PaginationItemsTag>
+
+        <Button size="sm" className={ css`flex-shrink: 0;` } onClick={ this.onChange(page + 1) } color="neutral" variant="outlined" squared disabled={ page >= numberOfPages }>
+          <Icon name="ChevronRight" size="xs" />
+        </Button>
+
+        <If condition={ !!showSizeChanger }>
+          <Select
+            className={ css`width: 64px; margin-left: 12px;` }
+            value={ pageSize }
+            options={ this.getPageSizeOptions(pageSizeOptions) }
+            onChange={ this.onChangePageSize }
+          />
+        </If>
+
+        <Text className={ css`margin-left: 12px;` }>
+          { numeral((page - 1) * pageSize + 1).format('0,0') } - { numeral(Math.min(total, page * pageSize)).format('0,0') } of { numeral(total).format('0,0') } records
```

Comment:
Why don't you use `Int.NumberFormat`?

This formatter does the same:
```
const formatter = new Intl.NumberFormat('en-EN', { useGrouping: true, maximumFractionDigits: 0 })
```

Reply:
@vorobeez good idea